app/views/posts/index.html.erb

```diff
@@ -0,0 +1,22 @@
+
+<div class="text-center">
+  <h1>Welcome to Members Only</h1>
+</div>
+<div class="row">
+
+  <div class="col-md-12">
+    <ul class="list-group">
+      <% @posts.each do |post| %>
+          <li class="list-group-item text-center">
+            <%= link_to post.title, post %>
+            <% if user_signed_in? %>
+            <small>
+              Posted by @<%= post.user.name %>
+            </small>
+          </li>
+            <% end %>
```

Comment:
This logic should be moved to a helper file. 
There should be absolutely NO logic in views. This is because views are only for displaying information that became available to it from its corresponding controller not generating it.

Reply:
gg wp